doc/on_chain_spec.md

```diff
@@ -669,3 +669,181 @@ that distributing their respective portions of the auction fee
 can be done in a single transaction.
 Later on, this can be generalized in a straightforward way
 to accommodate incremental fee distribution to a larger number of delegates.
+
+# Extensions
```

Comment:
I think splitting it out in different sections makes it easier to understand and follow.

These changes are not necessary to run an auction, but can be viewed as extra bits we add on top. Keeping the spec separate also allows people in the future to only pick what they need from it.

As for `ApprovedBidders`, if we merge this then we should definitely eliminate it, but otherwise we can just leave it there, as the previous spec worked with that

Reply:
> I think splitting it out in different sections makes it easier to understand and follow.

Why? You are saying same thing twice and you separate this from tx specs. All our doc is based on tx specs and requirements and that is the only case then tx specs do not say what tx does.

> These changes are not necessary to run an auction, but can be viewed as extra bits we add on top.

What you mean by "not necessary to run"? Our code does not work without it, it cannot be disabled. Without approval one cannot bid.

That is how our code works and spec should describe how our code works.

> but otherwise we can just leave it there, as the previous spec worked with that

It worked and now it does not work. This is mismatch between spec and code.